workshop/Zadanie 2/README.md

```diff
@@ -0,0 +1,62 @@
+8. Chcemy teraz, aby dodatkowy paragraf schował się po kliknięciu w przycisk, który ma napis `Mniej`. Pamiętajmy, że to cały czas ten sam przycisk. Użyjemy prostego warunku `if`, który pozwoli nam sprawdzić, czy zmienna `textArea` jest różna od `null`. Kiedy będzie ona równa `null` oznaczać to będzie, że ma `display` ustawiony na `none`, skrypt nie widzi jej w kodzie HTML. Ciało funkcji będzie wyglądąło mniej więcej tak:
+```javascript
+    var textArea = this.previousElementSibling // przypisujemy do zmiennej paragraf
+
+    if (textArea !== null) {
+        // Ten kod powinien wykonać się, gdy nasza zmienna nie równa się `null` tj. gdy ma jakąś zawartość. Zmienna nie będzie równa `null`, gdy będzie wyświetlona na stronie, tzn. jej `display` będzie równał się `block
+    } else {
+        // Ten kod wykona się, gdy nasza zmienna nie będzie widoczna na stronie, czyli w przypadku, gdy nie ma jej w kodzie HTML, bo ma `display` ustawiony na `none`
+    }
+```
+9. Wpisz odpowiedni kod do warunków. Pamiętaj, że w przypadku gdy zmienna nie jest wyświetlona i ustawiamy jej `display` na `block`, musimy ustawić też odpowiednią treść przycisku. Ponieważ nie wróci ona sama do `Więcej`, gdy kilkukrotnie klikniemy na przycisk.
+10. Teraz wszystko powinno dobrze działać dla pierwszego przycisku. Chcemy jednak, by dla dwóch pozostałych przycisków też pokazywały się odpowiednie paragrafy. Możemy skopiować kod i podłączyć funckję trzy raz do różnych przycisków, ale byłoby to bezsensowne powtarzanie kodu. Dlatego użyjemy pętli `for` do podłączenia funkcji do wszystkich przycisków jednocześnie. Dzięki odwołaniu się do `this` wewnątrz funkcji, zostanie wybrany odpowiedni paragraf dla danego przycisku.
+11. Najpierw przypiszmy całą funkcję, która była wywoływana przez `click` do funkcji, która nie będzie anonimowa. Wcześniej nasza funckja nie miała nazwy, po prostu od razu wypisywaliśmy jej ciało. Teraz zrobimy to w trochę inny sposób:
+```javascript
+    function showHide(): {
+        // tutaj skopiuj ciało funkcji
+    }
+```
+12. Teraz użyjemy pętli `for` do podłączenia funkcji `showHide` do wszystkich przycisków. Przyciski przechowujemy w zmiennej `buttons`, która jest tablicą. Nasza pętla przejdzie przez wszystkie elementy tablicy i na każdym z nich wykona wskazaną przez nas funkcję. Nasz kod będzie teraz wyglądał tak:
+```javascript
+    // Zaczynamy od pierwszego elementu tablicy, więc zmienną i ustawiamy na 0, bo to indeks pierwszego elementu
+    // Kończymy naszą pętlę, gdy i jest równe długości tablicy, tj. przeszło przez wszystkie jej elementy
+    // Nasza zmienna i ma podnosić się o 1 z każdą iteracją - oznaczamy to przez ++. Jest to równe wyrażeniu i + 1
+    for (var i = 0; i <= buttons.length; i++) {
+        buttons[i].addEvenetListener('click', showHide)
+    }
+```
```

Comment:
NIe wiem czy to im nie zamiesza (wkładanie do funkcji kodu zmieniającego widoczność). Nie wiem czy nie zostawiłbym to że na i-tym elemencie tablicy dodają po prostu event listenery.